### PR TITLE
Update deprecated packages

### PR DESCRIPTION
**Pull request by hsubramanianaks**
This PR is to remove deprecated packages like typemoq and replace them with sinon.js and chai for expects statements instead of assert statements, rearranged the existing UT's. 

Now deprecation warning has come down to two from several and for telemetry it is taken care in another PR. It is just uudi deprecated warning which needs attention.

image

Fixes https://github.com/Azure/vscode-bridge-to-kubernetes/issues/46

### CODE REVIEW

File: src/tests/suites/Logger-Test.ts
Comment (davidgamero) [inline src/tests/suites/Logger-Test.ts:5]: it's unfortunate that sinon doesn't support es6 imports

**Review by davidgamero — APPROVED** [review]
lgtm

File: src/tests/suites/RedactUtility-Test.ts
Comment (Tatsinnit) [inline src/tests/suites/RedactUtility-Test.ts:46]: 💡 idea / suggestion: please note and if there is a reading for use of riwil use here, instead of using -   ‘ expect(true).to.be.true;’  https://www.chaijs.com/api/bdd/#true

File: src/tests/suites/StringUtility-Test.ts
Comment (Tatsinnit) [inline src/tests/suites/StringUtility-Test.ts:18]: 💡🛠️ suggestion only please expect(true).to.be.true; - [link](https://www.chaijs.com/api/bdd/#true)

**Review by Tatsinnit — APPROVED** [review]
Thank you for this and adding me for review ❤️🙏 I have added specific comment and the chaijs doc recommended way of true check. Rest, Looks good.

before merging let’s make sure that we get this testing sign-off via our regular channels. ☕️💡

File: src/tests/suites/Logger-Test.ts
Comment (hsubramanianaks) [reply src/tests/suites/Logger-Test.ts:5]: I looked into it, there are issues open for it but didn't find them working yet.

File: src/tests/suites/StringUtility-Test.ts
Comment (hsubramanianaks) [reply src/tests/suites/StringUtility-Test.ts]: done updated.

File: src/tests/suites/RedactUtility-Test.ts
Comment (hsubramanianaks) [reply src/tests/suites/RedactUtility-Test.ts]: done, updated.

File: src/tests/suites/Logger-Test.ts
Comment (hsubramanianaks) [reply src/tests/suites/Logger-Test.ts]: done, updated.

**Review by Vidya2606 — APPROVED** [review]
LGTM

**Pull request merged**
